code/modules/clothing/gloves/miscellaneous.dm

```diff
@@ -69,31 +69,141 @@
 	permeability_coefficient = 0.9
 	siemens_coefficient = 0.9
 
+//Special type of gloves. Alt click and you get some special nodrop gloves
+/obj/item/clothing/gloves/heldgloves
+	name = "gloves"
+	var/obj/item/weapon/heldglove/right_glove = /obj/item/weapon/heldglove
+	var/obj/item/weapon/heldglove/left_glove = /obj/item/weapon/heldglove
+
+/obj/item/clothing/gloves/heldgloves/examine(mob/user)
+	. = ..()
+	to_chat(user, "<span class='notice'>Alt-click the gloves when worn to strap them into your hands.")
+
+/obj/item/clothing/gloves/heldgloves/unequipped(mob/unequipper, slot)
+	. = ..()
+	remove_gloves(unequipper)
+
+//We use alt-click to activate/deactive the gloves in-hand
+/obj/item/clothing/gloves/heldgloves/AltClick(mob/user)
+	var/mob/living/carbon/human/wearer = user
+	if(wearer.gloves != src) //We have to be wearing the gloves first
+		return
+
+	if(!remove_gloves(user))
```

Comment:
that's unintuitive, remove gloves should return true if you can and did remove gloves